### PR TITLE
[forge] Ability to set onchain config in tests

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR enables setting on-chain consensus config in Forge tests as part of the genesis config.

### Test Plan

Test in Forge

### CODE REVIEW

File: terraform/helm/genesis/values.yaml [hunk 1]
```diff
@@ -31,16 +31,18 @@ chain:
...
 imageTag: testnet
 
 genesis:
   image:
     # -- Image repo to use for tools image for running genesis
-    repo: aptoslabs/tools
+    repo: us-west1-docker.pkg.dev/aptos-global/aptos-internal/tools
     # -- Image tag to use for tools image. If set, overrides `imageTag`
-    tag:
+    tag: 8597cbdff337833d70560b6705ff7eadbcf62111
```
Comment (sionescu) [inline terraform/helm/genesis/values.yaml:45]: Let's avoid hardcoding a SHA1 here.

Comment (ibalajiarun) [reply terraform/helm/genesis/values.yaml:45]: That's for local testing. I'll undo.

File: crates/aptos-genesis/src/config.rs [hunk 2]
```diff
@@ -103,6 +107,7 @@ impl Default for Layout {
             total_supply: None,
             employee_vesting_start: Some(1663456089),
             employee_vesting_period_duration: Some(5 * 60), // 5 minutes
+            on_chain_consensus_config: OnChainConsensusConfig::default(),
```
Comment (sitalkedia) [inline crates/aptos-genesis/src/config.rs:110]: Can you add support for `OnChainExecutionConfig` as well ? :) that way we can just remove some ugly overrides @msmouse has for large TPS test to run...

Comment (ibalajiarun) [reply crates/aptos-genesis/src/config.rs:113]: Sure, done.

**Review by sitalkedia — APPROVED** [review]

File: terraform/helm/genesis/values.yaml (hunk 1 above)
Comment (msmouse) [reply terraform/helm/genesis/values.yaml]: Thanks bosses!

File: crates/aptos-genesis/src/config.rs (hunk 2 above)
Comment (msmouse) [reply crates/aptos-genesis/src/config.rs:113]: Thanks bosses!

**Review by msmouse — APPROVED** [review]

**Pull request merged**